src/sentry/rules/conditions/event_frequency.py

```diff
@@ -257,16 +257,15 @@ def query_hook(self, event, start, end, environment_id):
                     start=end - timedelta(minutes=60),
                     end=end,
                     filter_keys=filters,
-                    groupby=["bucketed_started"],
                     referrer="rules.conditions.event_frequency.EventFrequencyPercentCondition",
                 )
 
-            if result_totals["data"]:
-                session_count_last_hour = sum(
-                    bucket["sessions"] for bucket in result_totals["data"]
-                )
-            else:
-                session_count_last_hour = False
+            # Note: (RaduW) Kept sessions_count_last_hour = False (when there are no sessions) to maintain
+            #       compatibility with old the implementation. It doesn't make a lot of sense to me but
+            #       since the value is saved in cache I didn't want to change the original behaviour.
+            session_count_last_hour = (
+                result_totals["data"][0]["sessions"] if result_totals["data"] else False
```

Comment:
IIRC, we were storing it as `False` in order to differentiate between when we got no data back from Snuba, and when we got a count of 0. I think this was more relevant when we weren't making an hour long bucket and dividing it by the interval (this was an optimization done after the original implementation - we used to query per interval but this ends up making a lot more queries and puts a lot more data in the cache). Just looking at the code, it seems like it would be fine if we didn't keep `False` now...so maybe be okay to change.